octopusdeploy_framework/resource_space.go

```diff
@@ -325,7 +332,15 @@ func (s *spaceResource) cancelRunningTasks(ctx context.Context, spaceID string)
 		return fmt.Errorf("failed to get running tasks while checking: %w", err)
 	}
 
-	if len(remainingTasks.Items) > 0 {
+	var remainingRunningTasksInSpace []*tasks.Task
```

Comment:
This is basically a duplicate of lines 305:311, so can extract this into a common function. Just want to get feedback on the approach first.

Reply:
We could retry here, but a user can also retry the apply, so I'm not sure if its worth it.